host/elf.c

```diff
@@ -510,6 +510,54 @@ int elf64_find_symbol_by_name(
     return rc;
 }
 
+int elf64_get_dynamic_symbol_table(
+    const elf64_t* elf,
+    const elf64_sym_t** symtab,
+    size_t* size)
+{
+    int rc = -1;
+    size_t index;
+    const elf64_shdr_t* sh;
+    const char* SECTIONNAME = ".dynsym";
+    const elf64_word_t SH_TYPE = SHT_DYNSYM;
+
+    if (!_is_valid_elf64(elf) || !symtab || !size)
+        goto done;
+
+    *symtab = NULL;
+    *size = 0;
+
+    /* Find the symbol table section header */
+    if ((index = _find_shdr(elf, SECTIONNAME)) == (size_t)-1)
+        goto done;
+
+    if (index == 0 || index >= _get_header(elf)->e_shnum)
+        goto done;
+
+    /* Set pointer to section header */
+    if (!(sh = _get_shdr(elf, index)))
+        goto done;
+
+    /* If this is not a symbol table */
+    if (sh->sh_type != SH_TYPE)
+        goto done;
+
+    /* Sanity check */
+    if (sh->sh_entsize != sizeof(elf64_sym_t))
+        goto done;
+
+    /* Set pointer to symbol table section */
+    if (!(*symtab = (const elf64_sym_t*)_get_section(elf, index)))
+        goto done;
+
+    /* Calculate number of symbol table entries */
+    *size = sh->sh_size / sh->sh_entsize;
+
+    rc = 0;
+done:
+    return rc;
+}
```

Comment:
You may want to add logging info, if appropriate.

Reply:
This is based on existing code. So will keep it the same logging verbosity for now.
We can revisit it later.